cmake/modules/verify-toolchain.cmake

```diff
@@ -15,6 +15,8 @@
 # it takes the following arguments:
 # FORMAT=json: Print the output as a json formatted string, useful for Python
 
+include_guard(GLOBAL)
+
```

Comment:
Missing docs for required variables, ARCH, ZEPHYR_TOOLCHAIN_VARIANT.

Alternatively, we could keep this outside of the modules directory, as normal .cmake files don't need to document their dependencies.

Reply:
> Missing docs for input variables, ARCH, ZEPHYR_TOOLCHAIN_VARIANT.
>
> Alternatively, we could keep this outside of the modules directory, as normal .cmake files don't need to document their dependencies.

This is an existing cmake file.

This PR doesn't aim at solving short-comings in our existing CMake documentation.

Neither is the location of a CMake file under `cmake` vs . `cmake/modules` decisive as to whether or not a CMake file should be documented.
The fact that cmake files today are not properly documented, doesn't mean we should do so for new files regardless of their location.
Just like we document python scripts (some scripts better than others).

The reason why descriptions are added to files such as:
https://github.com/zephyrproject-rtos/zephyr/blob/4387c0aaee6cdbb5e1a65748f9e912ac2ae85972/cmake/modules/root.cmake
https://github.com/zephyrproject-rtos/zephyr/blob/4387c0aaee6cdbb5e1a65748f9e912ac2ae85972/cmake/modules/arch.cmake
https://github.com/zephyrproject-rtos/zephyr/blob/4387c0aaee6cdbb5e1a65748f9e912ac2ae85972/cmake/modules/boards.cmake
etc.

are because those are new / repurposed files, and as such they should be properly documented when added to the repository.

Cleaning up existing files is something that can be considered covered by: #9947
This PR is just a first step in improving the build system, where current build system behavior is not changed, as also mentioned here: https://github.com/zephyrproject-rtos/zephyr/pull/41301#discussion_r790009740

Reply:
Which modules in cmake/modules is it supported that users can include, and which are considered internal scripts?

Reply:
This PR lays down the foundation for a more modular approach targeting the Zephyr specific use-cases mentioned with this PR: https://github.com/zephyrproject-rtos/zephyr/pull/41301#issue-1083225661

So as such, `find_package(Zephyr)` is still the official way to load Zephyr build system as it has always been.

However, just like users today may patch / reuse Zephyr code as they like, they can of course start to use the new approach as the find fit.

But until additional cleanup of internal module code, and proper documentation has been added, similar to what we provide with `find_package(Zephyr)`, you may consider everything internal scripts, just as they are today.

Reply:
I misunderstood then, since the PR description said it was possible to include individual modules I assumed it was also supported.

Reply:
> said it was possible to include individual modules

Description says, second paragraph:
https://github.com/zephyrproject-rtos/zephyr/pull/41301#issue-1083225661
> so that it can easier be reused through the Zephyr package mechanism.

so it can re-use and include individual modules, but through `find_package(Zephyr)`.
There are two clear internal use-cases for this:
#40555 
#41302